### DISCUSSION TITLE
How to change the locale and language pack at the same time ?

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#11172] [General - Components / Directives / etc] [answered]
Hi, i want to change the locale in app internationalization and also change the language pack at the same time. I followed both examples in: [Here](https://quasar.dev/options/app-internationalization#create-language-switcher) and [Here](https://quasar.dev/options/quasar-language-packs#change-quasar-language-pack-at-runtime)

Now i want to merge them together to create a language switcher to change the locale and the language pack at the same time like this:

```vue
<template>
  <q-select
    class="language-switcher"
    v-model="locale"
    :options="localeOptions"
    borderless
    emit-value
    map-options
    options-dense
    color="black"
  />
</template>

<script>
import { ref, watch, computed } from "vue";
import { useI18n } from "vue-i18n";
import { useQuasar } from "quasar";

export default {
  setup() {
    const $q = useQuasar();
    const { locale } = useI18n({ useScope: "global" });
    const localeOptions = [
      { value: "en-US", label: "English" },
      { value: "ar", label: "العربيه" },
    ];

    watch(locale, (val) => {
      import(
        /* webpackInclude: /(en-US|ar)\.js$/ */
        "quasar/lang/" + val
      ).then((val) => {
        $q.lang.set(val);
      });
    });

    return {
      locale,
      localeOptions,
    };
  },
};
</script>
```

However this did not work and i am stuck. Notice that i only want to change the language pack cuz arabic is written from right to left and i wanna change the direction by changing the language pack used. I would appreciate any help, thank you.

**Comment by yusufkandemir** [comment, marked as answer]
Notice the `.default` usage on [here](https://quasar.dev/options/quasar-language-packs#change-quasar-language-pack-at-runtime): `$q.lang.set(lang.default)`

So, you would need to do
```diff
- $q.lang.set(val)
+ $q.lang.set(val.default)
```

**Reply by ghost** [reply]
Thank you so much
